packages/kit/src/utils/streaming.js

```diff
@@ -23,22 +23,29 @@ function defer() {
  * }}
  */
 export function create_async_iterator() {
-	let deferred = defer();
+	let deferred = [defer()];
 
 	return {
 		iterator: {
 			[Symbol.asyncIterator]() {
 				return {
-					next: () => deferred.promise
+					next: async () => {
+						const next = await deferred[0].promise;
+						deferred.shift();
```

Comment:
better safe than sorry - this adheres more closely to the iterator protocol
```suggestion
						if (!next.done) deferred.shift();
```